simplicity-sys/src/ffi.rs

```diff
@@ -0,0 +1,480 @@
+// Simplicity Bindings
+// Written in 2022 by
+//   Christian Lewe <clewe@blockstream.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! FFI Bindings
+//!
+//! This module contains all bindings to the C library types and functions.
+//! It is split into several modules, each one corresponding to a `.h` file
+//! in the C library.
+//!
+//! All types are converted to CamelCase and prefixed with the letter C;
+//! function names are unchanged.
+
+use std::fmt;
+
+/// Used with `evalTCOProgram` to enforce consensus limits.
+pub const BUDGET_MAX: ubounded = 4000050;
+
+/// Used in the C code for various purposes; actually defined as `uint_least32_t`
+/// which seems not to have an equivalent in Rust libc.
+// There are sanity checks for both these types in c_jets/frame_ffi.rs
+#[allow(non_camel_case_types)]
+pub type ubounded = usize;
```

Comment:
This is correct as `u32`. I only test the `UWORD` as `usize`. Apparently, `uint_least32_t` maps to u32, while `uint_fast32_t` maps to `usize`. 

The tests are locally failing when I do this. This also means that we are not correctly running this in CI. Can you also add the check on `ubounded` in `c_jets::c_env::tests::test_sizes` ?

Reply:
The tests were passing because
* We were directly testing `usize` instead of the `UWORD` type alias, so the test would've passed no matter what I'd put for `UWORD`
* We weren't testing `ubounded` at all.

When I address both those issues then I see test failures, and I have to correctly set the types for the tests to pass. So I think we're ok.

Reply:
I guess we should also add tests for all the new structures. I'll add that in a separate commit.